src/test/streams_tests.cpp

```diff
@@ -101,8 +101,17 @@ BOOST_AUTO_TEST_CASE(streams_vector_reader)
     signed int d;
     BOOST_CHECK_THROW(reader >> d, std::ios_base::failure);
 
+    BOOST_CHECK(reader.empty());
+    BOOST_CHECK_THROW(reader.seek(-7), std::ios_base::failure);
+    BOOST_CHECK(reader.empty());
```

Comment:
This is redundant with the `BOOST_CHECK_EQUAL(reader.size(), 0)` two lines down. I think this one can be removed.